scene/gui/text_edit.cpp

```diff
@@ -4333,6 +4333,16 @@ bool TextEdit::is_right_click_moving_caret() const {
 	return right_click_moves_caret;
 }
 
+void TextEdit::set_selection_mode(Selection::Mode p_mode, int p_line, int p_column) {
+	selection.selecting_mode = p_mode;
+	if (p_line >= 0) {
```

Comment:
Should check it's not greater then the text size with `ERR_FAIL_INDEX(p_line, text.size());`